cmd/core/main.go

```diff
@@ -151,3 +163,45 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+// setupChecks is used to create readiness&liveness probes
+func setupChecks(mgr ctrl.Manager) {
+	setupLog.Info("creating readiness/health check")
+	if err := mgr.AddReadyzCheck("ping", healthz.Ping); err != nil {
+		setupLog.Error(err, "unable to create ready check")
+		os.Exit(1)
+	}
+
+	if err := mgr.AddHealthzCheck("ping", healthz.Ping); err != nil {
+		setupLog.Error(err, "unable to create health check")
+		os.Exit(1)
+	}
+}
+
+// waitSecretVolume waits for webhook secret ready to avoid mgr running crash
+func waitSecretVolume(certDir string) {
+	timeout := 300 * time.Second
```

Comment:
I don't think we need so much long time, the k8s itself has mechanism to restart. As I know, this function we added here is just to let the controller won't make the K8s restart backoff take too long. 

I think 60s is enough, right?

Reply:
Yes, that's right. But maybe 90s is okay. After testing several times, the average time is around 60-70s.